src/counter.cpp

```diff
@@ -440,24 +467,41 @@ ApproxMC::SolCount Counter::calc_est_count()
     ApproxMC::SolCount ret_count;
     if (num_hash_list.empty() || num_count_list.empty()) return ret_count;
 
-    // round model counts
     if (num_hash_list[0] > 0) {
         double pivot = 9.84*(1.0+(1.0/conf.epsilon))*(1.0+(1.0/conf.epsilon));
 	    for (auto cnt_it = num_count_list.begin(); cnt_it != num_count_list.end(); cnt_it++) {
-            if (conf.epsilon < sqrt(2)-1) {
+            // Adjust model counts of ApproxMC7 to improve confidence.
+            // See Line 15 of Algorithm 4 in AAAI25 paper "Towards Real-Time Approximate Counting"
+            // https://ojs.aaai.org/index.php/AAAI/article/view/33231
+            // The usage of sqrt(2*alpha/beta) is justified in the proof of Lemma 5, specifically, Equation 7 and 10.
+            if (conf.epsilon >= conf.appmc7_eps_cutoff) {
+                *cnt_it *= sqrt(2*alpha/beta);
```

Comment:
As mentioned in the other comment, please add an assert above this to make sure it's never -1. I guess it cannot be -1.